src/ansys/dpf/core/data_sources.py

```diff
@@ -69,13 +77,21 @@ class DataSources:
     Initialize a model from a result path.
 
     >>> from ansys.dpf import core as dpf
-    >>> my_data_sources = dpf.DataSources('file.rst')
+    >>>
+    >>> # Create the DataSources object with a main file path
+    >>> my_data_sources = dpf.DataSources(result_path='file.rst')
+    >>> # Get the path to the main result file
     >>> my_data_sources.result_files
     ['file.rst']
 
     """
 
-    def __init__(self, result_path=None, data_sources=None, server=None):
+    def __init__(
+        self,
+        result_path: Optional[Union[str, os.PathLike]] = None,
+        data_sources: Optional[dpf.DataSources] = None,
```

Comment:
Not a `dpf.DataSources`.
A `Union` of either `int` for a memory address or `ansys.grpc.dpf.data_sources_pb2.DataSources` as in the original signature.